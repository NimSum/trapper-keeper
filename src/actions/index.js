export const addNotes = notes => ({
  type: 'ADD_NOTES',
  notes
});

export const addNote = note => ({
  type: 'ADD_NOTE',
  note
})

export const updateNote = note => ({
  type: 'UPDATE_NOTE',
  note
})
