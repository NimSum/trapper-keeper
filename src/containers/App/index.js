import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addNotes } from '../../actions/';
import CardContainer from '../../components/CardContainer';
import { Route } from 'react-router-dom'
export class App extends Component {

  async componentDidMount() {
    const response = await fetch('http://localhost:3000/api/v1/notes');
    const notes = await response.json();
    this.props.setNotes(notes.notes)
  } 

  render() {
    return (
      <div>
        <h1>Trapper Keeper</h1>
        < CardContainer />
        <Route path='/new-note' component={Form} />
      </div>
    )
  }
}

const mapStateToProps = (state) => ({

})

const mapDispatchToProps = (dispatch) => ({
  setNotes: (notes) => dispatch(addNotes(notes))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
