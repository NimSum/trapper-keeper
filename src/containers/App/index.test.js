import React from "react";
import { App } from "./index";
import { shallow } from "enzyme";
import { mapStateToProps, mapDispatchToProps } from "../App";
import { addNotes } from "../../actions/index";

describe("App", () => {
  let wrapper;
  let mockNotes;
  let mockSetNotes = jest.fn();
  
  beforeEach(() => {
    wrapper = shallow(
      < App 
        setNotes={ mockSetNotes } />
    );
    mockNotes = [
      {
        title: "randomnote",
        id: "1",
        listItems: [
          {
            id: "1",
            body: "asdf",
            completed: false
          }
        ]
      },
      {
        title: "randomnoteTWO",
        id: "2",
        listItems: [
          {
            id: "2",
            body: "asdf",
            completed: false
          }
        ]
      },
      {
        id: "21f67edd-ec5a-4ab4-8b9e-18fa1326a70e",
        title: "Angry Man",
        listItems: [
          {
            id: "92f3600b-00e5-4551-82c3-a1c7b409d9d2",
            body: "Yelling in a coffee shop",
            completed: false
          }
        ]
      }
    ];
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });



  describe("mapStateToProps", () => {
    it("should return an array or notes", () => {
      const mockState = { notes: mockNotes };
      const expected = { notes: mockNotes };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe("mapDispatchToProps", () => {
    it("should call dispatch with a note action on componentDidMount", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addNotes(mockNotes);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setNotes(mockNotes);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
