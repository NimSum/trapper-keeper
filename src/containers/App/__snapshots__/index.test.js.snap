// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<main>
  <h1>
    Trapper Keeper
  </h1>
  <hr
    className="header-break"
  />
  <Link
    exact={true}
    to="/new-note"
  >
    <h2>
       
      Add a New Note 
      <i
        className="fas fa-plus-square"
      />
       
    </h2>
  </Link>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Form)",
        "type": [Function],
      }
    }
    path="/new-note"
  />
  <Route
    path="/notes/:id"
    render={[Function]}
  />
  <ConnectFunction />
</main>
`;
